### PR TITLE
Add command for configuring basic attributes in transactions subsystem.

### PR DESCRIPTION
**Pull request by istraka**

### CODE REVIEW

File: commands/src/main/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicTransactionAttributes.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        }
+        if (objectStoreRelativeTo != null) {
+            batch.writeAttribute(transatcionsAddress, "object-store-relative-to", objectStoreRelativeTo);
+        }
+
+        ops.batch(batch);
+
+    }
+
+    public static final class Builder<THIS extends Builder> {
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicTransactionAttributes.java]: Do you expect to have subclasses? If not, the `THIS` machinery is useless (and a bit confusing).

File: commands/src/main/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicTransactionAttributes.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    private Boolean processIdUuid;
+    private String processIdSocketBinding;
+    private Integer processIdSocketMaxPorts;
+
+    private String socketBinding;
+    private String statusSocketBinding;
+    private Boolean recoveryListener;
+
+    private String objectStorePath;
+    private String objectStoreRelativeTo;
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicTransactionAttributes.java]: Why some `final` and some not?

**Comment by istraka** [comment]
Issues have been fixed.

File: commands/src/main/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicTransactionAttributes.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
+package org.wildfly.extras.creaper.commands.transactions;
+
+
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicTransactionAttributes.java]: Just curious, why do you always have 2 empty lines between `package ...` and the first `import ...`? Looks weird.

Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicTransactionAttributes.java]: Other than that, the PR looks fine.

**Comment by istraka** [comment]
2 empty lines has been removed. Commits have been squashed.

**Comment by Ladicek** [comment]
OK, thanks.

**Comment by Ladicek** [comment]
Please rebase.

**Comment by istraka** [comment]
Done

**Comment by Ladicek** [comment]
Did you run `mvn clean verify`? This commit seems to introduce a lot of test failures, mostly in the datasources-related tests.

**Comment by istraka** [comment]
Online test clean after itself now.

**Comment by Ladicek** [comment]
Why do you create the offline client twice? Creating it once should be enough.

**Comment by istraka** [comment]
It is created only once, now.

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicAttributesOnlineTest.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    @Test
+    @InSequence(1)
+    public void startServer() throws IOException, CommandFailedException {
+        controller.start(ManualTests.ARQUILLIAN_CONTAINER);
+        offlineClient = ManagementClient.offline(
+                OfflineOptions.standalone()
+                        .rootDirectory(new File(System.getProperty("user.dir"), "target/jboss-as"))
+                        .configurationFile("standalone.xml")
+                        .build());
+        offlineClient.apply(CONFIGURATION_BACKUP.backup());
```
Comment (Ladicek) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicAttributesOnlineTest.java]: I think that you should do the backup first, and only then start the server.

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicAttributesOnlineTest.java [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        result = ops.readAttribute(TEST_TRANSACTIONS_ADDRESS, "object-store-relative-to");
+        result.assertSuccess();
+        assertEquals("object-store-relative-to should not be changed", "relative-to", result.stringValue());
+    }
+
+    @Test
+    @InSequence(3)
+    public void stopServer() throws CommandFailedException, IOException {
+        offlineClient.apply(CONFIGURATION_BACKUP.restore());
+        controller.stop(ManualTests.ARQUILLIAN_CONTAINER);
```
Comment (Ladicek) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicAttributesOnlineTest.java]: Same as above -- first stop the server, then restore.

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicAttributesOnlineTest.java (hunk 4 above)
Comment (istraka) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicAttributesOnlineTest.java]: Done.

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicAttributesOnlineTest.java (hunk 5 above)
Comment (istraka) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/transactions/ChangeBasicAttributesOnlineTest.java]: Done.

**Comment by Ladicek** [comment]
OK, looks good. Please squash and rebase.

**Comment by istraka** [comment]
Done.

**Comment by Ladicek** [comment]
Hmm, the test seems to be failing on EAP 6 with `{"JBAS014653: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-4" => "JBAS014792: Unknown attribute statistics-enabled"}}`. Fairly easy to fix in online, but what about offline?

**Comment by istraka** [comment]
Fixed.

**Comment by Ladicek** [comment]
The tests are now passing in the `as7` and `wildfly10` profiles. They are still failing in the `wildfly8` and `wildfly9` profiles.

I typically don't run tests in these profiles when merging PRs, but this one has been a bit more problematic, so I decided to run the WildFly 8 and 9 profiles as well. Looks like my intuition didn't fail this time :-)

**Comment by istraka** [comment]
Name of attributes are correct according to server version.

**Comment by Ladicek** [comment]
Please put the note that the offline variant is EAP7-only to the _class_ javadoc. Then, rebase and squash; looks good to merge now.

**Comment by istraka** [comment]
Done.

**Pull request merged**